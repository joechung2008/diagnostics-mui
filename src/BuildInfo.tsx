import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
} from "@mui/material";

const BuildInfo: React.FC<BuildInfoProps> = ({ buildVersion }) => {
  const items = [
    {
      name: "Build Version",
      value: buildVersion,
    },
  ];

  return (
    <Table aria-label="Build Info">
      <TableHead>
        <TableRow>
          <TableCell>Name</TableCell>
          <TableCell>Value</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {items.map((item) => (
          <TableRow key={item.name}>
            <TableCell>{item.name}</TableCell>
            <TableCell>{item.value}</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  );
};

export default BuildInfo;
