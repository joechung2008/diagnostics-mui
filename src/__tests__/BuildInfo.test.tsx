import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import BuildInfo from "../BuildInfo";

describe("BuildInfo", () => {
  it("matches snapshot", () => {
    const { container } = render(<BuildInfo buildVersion="1.0.0" />);
    expect(container).toMatchSnapshot();
  });
});
