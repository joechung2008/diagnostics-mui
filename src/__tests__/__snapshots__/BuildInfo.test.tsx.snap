// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BuildInfo > matches snapshot 1`] = `
<div>
  <table
    aria-label="Build Info"
    class="MuiTable-root css-1xwxv7r-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
          scope="col"
        >
          Value
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          Build Version
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
        >
          1.0.0
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
